### DISCUSSION TITLE
hope support WeChat mini-games, Douyin mini-games, QQ mini-games, and Alipay mini-games

### DISCUSSION THREAD

**Discussion by huxinfeng** [godotengine/godot-proposals#10947] [Platforms]
### Describe the project you are working on

Recently we developed a game using Godot. The Android platform performed well. Now we want to publish it to the WeChat platform, but we encountered many problems;

The Godot engine can only export h5 format, which is not very suitable for the packaging format of WeChat mini games. I want to directly read the wasm file and create the game directly, but I encountered that the fetch and other APIs used by the index.js exported by godot cannot be used in WeChat. I have received support. Is there any good solution to enable Godot engine games to be published on the WeChat platform?

### Describe the problem or limitation you are having in your project

Cannot support WeChat mini-games, Douyin mini-games, QQ mini-games, and Alipay mini-games
wx-mini-games：https://developers.weixin.qq.com/minigame/dev/guide/game-engine/unity-webgl-transform.html

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

not

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

not

### If this enhancement will not be used often, can it be worked around with a few lines of script?

no

### Is there a reason why this should be core and not an add-on in the asset library?

I hope to be able to support China’s unique gaming environment. At present, Godot is not particularly big in the Chinese market. If it can support all small games, I believe many people will join Godot.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/discussions/7681.

**Reply by flyshuttle2** [reply]
yes, I believe so

**Comment by tomelsj** [comment]
Technically, this _should_ be possible to use godots web export and then some wrapping work. What we need to consider is that many Chinese platforms only support WebGL1, so no WebGL2, to not speak of WebGPU... Wechat is an exception; Tencent is a gaming company as well, so I believe they actually support WebGL2, at least to some degree.

**Reply by di404** [reply]
As far as I know, WeChat still doesn't have good support for WebGL2

**Reply by frank28** [reply]
That's not true. WeChat minigame has good WebGL2 support already. https://developers.weixin.qq.com/minigame/dev/guide/game-engine/unity-webgl-transform/Design/WebGL2.html https://developers.weixin.qq.com/minigame/dev/guide/performance/perf-high-performance-plus.html
